crates/sage-cli/src/input.rs

```diff
@@ -221,6 +231,7 @@ impl Input {
     fn check_tolerances(tolerance: &Tolerance) {
         let (lo, hi) = match tolerance {
             Tolerance::Ppm(lo, hi) => (*lo, *hi),
+            Tolerance::Pct(_, _) => unreachable!("Pct tolerance should never be used on mz"),
```

Comment:
I don't think we should reify this into an enum, if it should never be constructed. If pct tolerances are just used for IM, I recommend just keeping it as a standard float. Otherwise, create a second enum that is just Ppm/Pct/etc if ppm is also an option for IM tolerance (since Da doesn't seem to make sense).

Additionally, `unreachable` should never be used unless you can *prove* that it is truly unreachable. I would just use panic here, since I don't see any evidence of such proof.

Reply:
Well ... looking at it again I think the "abstraction" of tolerance could go both ways ... in theory the mass tolerance could be a percentage (although it would "feel" wrong to use). The same way I do agree that 'da' makes no sense for IM, but in theory the tolerance could also be renamed as 'absolute', where it would make sense to have it (and for backwards compatibility, it could be aliased/newtyped for deserialization from the config.

Reply:
also ... fixed the `unreachable` aspect of it. (decided to just allow it ... if someone specifies their tolerance as a percentage I expect them to know what they are doing :P)